graph/telemetry/istio/appender/workload_entry.go

```diff
@@ -0,0 +1,64 @@
+package appender
+
+import (
+	"github.com/kiali/kiali/business"
+	"github.com/kiali/kiali/graph"
+	"github.com/kiali/kiali/log"
+)
+
+const WorkloadEntryAppenderName = "workloadEntry"
+
+// WorkloadEntryAppender correlates trafficMap nodes to corresponding WorkloadEntry
+// Istio objects. If the trafficMap node has a matching WorkloadEntry, a label is
+// added to the node's Metadata. Matching is determined by the "app" and "version"
+// labels on both the trafficMap node and the WorkloadEntry object being equivalent.
+// A workload can have multiple matches.
+type WorkloadEntryAppender struct {
+	GraphType string
+}
+
+// Name implements Appender
+func (a WorkloadEntryAppender) Name() string {
+	return WorkloadEntryAppenderName
+}
+
+// AppendGraph implements Appender
+func (a WorkloadEntryAppender) AppendGraph(trafficMap graph.TrafficMap, globalInfo *graph.AppenderGlobalInfo, namespaceInfo *graph.AppenderNamespaceInfo) {
+	if len(trafficMap) == 0 {
+		return
+	}
+
+	log.Trace("Running workload entry appender")
+
+	a.applyWorkloadEntries(trafficMap, globalInfo, namespaceInfo)
+}
+
+func (a WorkloadEntryAppender) applyWorkloadEntries(trafficMap graph.TrafficMap, globalInfo *graph.AppenderGlobalInfo, namespaceInfo *graph.AppenderNamespaceInfo) {
+	for _, n := range trafficMap {
+		// Only a workload or app node can be a workload entry
+		if n.NodeType != graph.NodeTypeWorkload && n.NodeType != graph.NodeTypeApp {
+			continue
+		}
+
+		istioCfg, err := globalInfo.Business.IstioConfig.GetIstioConfigList(business.IstioConfigCriteria{
+			IncludeWorkloadEntries: true,
+			Namespace:              n.Namespace,
+		})
+		graph.CheckError(err)
+
+		log.Tracef("WorkloadEntries found: %d", len(istioCfg.WorkloadEntries))
+
+		for _, entry := range istioCfg.WorkloadEntries {
+			if labels, ok := entry.Spec.Labels.(map[string]interface{}); ok {
+				if labels["app"] == n.App && labels["version"] == n.Version {
```

Comment:
> But, more than this, I think we may want to make an optimization here because this is now the third appender to independently query for IstioConfig (see istio_details and service_entry). We may want to combine the three requests into one call that has:

I'm happy to make this change but aren't these objects already cached by the business layer? If that's the case then querying for them should still just be pulling them from memory?

Reply:
It's possible, I'm not sure if we pull all of the different information that can be specified in the ConfigCriteria.  when the NamespaceInfo was originally introduced there was no cache in Kiali, and so when we did introduce caching we ended up with some "double-caching", and some, maybe all, of that has been removed.  If you can prove that the information being requested is cached then we can ignore this suggestion.  Also, in that case, it could be useful to add a comment reflecting that the calls are quick due to caching.

Reply:
It looks like pulling from the cache is happening [here](https://github.com/kiali/kiali/blob/master/business/istio_config.go#L270-L272) in the business layer and I see some logs indicating cache hits that correspond to the workload entry query:
```
2021-08-18T19:12:02Z TRC [Kiali Cache] Get [resourceType: serviceentries] for [namespace: bookinfo] =  1                                                                                      
2021-08-18T19:12:02Z TRC Running workload entry appender                                                                                                                                      
2021-08-18T19:12:02Z TRC [Kiali Cache] Get [resourceType: workloadentries] for [namespace: bookinfo] =  1                                                                                     
2021-08-18T19:12:02Z TRC WorkloadEntries found: 1                                                                                                                                             
2021-08-18T19:12:02Z TRC [Kiali Cache] Get [resourceType: workloadentries] for [namespace: bookinfo] =  1                                                                                     
2021-08-18T19:12:02Z TRC WorkloadEntries found: 1                                                                                                                                             
2021-08-18T19:12:02Z TRC Found matching WorkloadEntry                                                                                                                                         
2021-08-18T19:12:02Z TRC [Kiali Cache] Get [resourceType: workloadentries] for [namespace: bookinfo] =  1                                                                                     
2021-08-18T19:12:02Z TRC WorkloadEntries found: 1                                                                                                                                             
```

Reply:
OK, great, then we should be good just relying on the cache.  You can ignore my comment.